Source/WebKit/WebProcess/GPU/graphics/RemoteResourceCacheProxy.cpp

```diff
@@ -196,6 +196,8 @@ void RemoteResourceCacheProxy::clearImageBufferBackends()
 
 void RemoteResourceCacheProxy::clearDecomposedGlyphsMap()
 {
+    for (auto& decomposedGlyphs : m_decomposedGlyphs.values())
+        decomposedGlyphs->removeObserver(*this);
```

Comment:
`m_decomposedGlyphs` holds `WeakPtr`s. Do we need to do `null` checks here?

Is `this` guaranteed to be the only observer?

The commit message doesn't have any explanation about why the current behavior of `DecomposedGlyphs::~DecomposedGlyphs()` isn't sufficient.

Can this be tested?

Reply:
> `m_decomposedGlyphs` holds `WeakPtr`s. Do we need to do `null` checks here?

We don't because the destructor for the `CanMakeWeakPtr` base class of `DecomposedGlyphs` runs after the `DecomposedGlyphs` destructor, and the latter will drop the `WeakPtr`s.

> Is `this` guaranteed to be the only observer?

No, since different documents could pick the same entry out of the global `GlyphDisplayListCache`.

> Can this be tested?

I will think about that.